### PR TITLE
Mouse touchscreen form

### PR DESCRIPTION
**Pull request by david-roper**
Version 1.0 of the mouse touchscreen form
Closes issue #5 

form can be seen [here](https://playground.opendatacapture.org/?files=NobwRAxg9gdgLgU3mAXGA9AKgAQIM4A2AlvALQAmReAhgEYELYAOCATgGYIRxGxVzo8UVnFJRaAKy5w82TOgA6MJdBh442ENnIJ2JBAEk1cVgFcAtkg0BfbAF5s1AO7UiGouabC4ACgDk6GbwHgjoAG4AjOgAAlAsMOTUcNQQ1ExwpqyhQTyWpNBZ6CQ6AB4AdBJ4fgCUANwqsOqa2ABe2LYOzq7unt7%2BgabBluFRLVDk0QDMZQBM0yVFCQjllTX1yjDLXiLautSmBBo6eptG6maW8D4gStjYANbFKNh%2BAGIA8gBKALJ%2BADS3bAEagwADmpmooIQzz8SH%2BgOSoLwz2Afm%2BUFMeAQf2wABUMRAABZ4CBZOF-PwABQAagARf5%2BACs%2BUJUCIEAQfgAugCYHcSIhWDBqARnjc%2BXdcJQeLBnhFeZLsMLLDDvu8AKoAZQAogB9XEagDCAAlNYbPtrtQA5XUfH5%2BQHWBXYVSIeBiwF3LZsELwXEATxYHoliseCRh5xIoPhIclYWorCIILgMNY1EoUBjiruwNoCFF2AUYADLGwUHY2DgBOJpIQSGwWLweF4yjAzsVcRlahQ4uzkqLNNpRZQA7p2B8lNcrCcVEYtKopI8JCSLeqRfb2aLzKJbI5w8LYEZhs1n1xuPHjJdrPZjCxiZF2CyKS7lZCleoeHua7Ansl1l-Tq-t6iaXHAmqItCmi-ncYbkBGJhRlm2bxve7ovFiDDcEhiq5vmzxFtqJQsCBVgNhB67QWW6QtsiUGxoqRaEnQbgQl2w5FsazEZCusAUfR-ZgExtAsTxMAzNQ7FgJxwncV24l8X2dyMVxrEtjMtCSdJIlyRpbaUUpYACkmBCVtW%2B5FkYbjGaZphEgpilFuYmIaFWtmEuZYDfM5Nl2Xp-EGU5TQkFZSQIB5XlBTAIWIPZfZFkwgxUIS2AkAUWTcJJlKJXgyWpcI6VwLFm5gOwBwmY2za8agB6vGVDb4JVraUf%2BsaAbGTBhOQ4GQpBvbZrBMLkP6yrsthko6EwtGosBvpwCWnI4rCRE%2BqB3VQtyG53FkSysD4iTJNUfV9kQ7B7Uk1BlDNoHzfYdgOKOQ5gNgABkz3aOdl3LSR8BrYwd33SVdUVS2RaHfpkpZBkQp0Yp-VPOhCFgmNsMoUmaF%2BGmGbI4puEFg9uBfbNZE9UVsOdjRwaw3FYDUAAjqYVBWVV%2BFgAAgvTjNsX5VPFVkYRsDQBCSZ8CD86wguk1TRZZKQfMCyKHki7Lovy0LP7%2BdmLU81rik632kOZHyMBlc1AEbno-OGteHK-ZTcPhi8Q0jRA2MTVNS3EbN83-C8V1WL9G2UdtOi7ft1BgxrJ1nckn2e9dgZ-XdB7btbYVPa970x37P0QbdAOlQQ5UNSDYARzzBvQ0dVMDQjiZI5timo8mqbprw2N9rjLMp7ujDZxo6gk9zPNUV2tFVzzRYACz1ao5BkR4ByYh509YrP89OQQS9D8PB4zDPsBz%2BoC%2Bb3gHl76vB-r4vp-b8PA6sFAlb3mrLOUg-T-GUW4OKnruvg7-ioK5GxNvxX%2BbVFREmoOYPMrArQWBgXbUM8M-DG2gWwbGTd0YkASnAbGncXhWygTApU8C2DjgiKQCIMxqhjXAZKSBaDWCah9CKQ04xeqURrn4SM9dKKYJTC8bBphcEN3wX4QhjD6rPxdOwh0rUNzmCIAQe4OVqD3AQAAITTA7ceDxkE8OjA3fhLcsaiLoHhA83wlEqKYuo7AtBtHkEluTRoiDipWgQEooQfJwJpicDA1g-psBWOUao9RkkPFeNgNgXxzgAlBJCTYtRacG4GXeHAQkbAPLpMyawL%2BoCza-kUaE2xmjHE5LYG4mCyCnZQNGg3N2KI-DFKSeorRIJyCB34sHNg0dw66P5KdMOZQWlhLKR0vOLwKmsBqAMiGCAoZ8jmUgh23DEaGO-nGBMaMBFFiEYVW%2BsMxHtISNgK0UDOSbIAX%2BcGQCSGF1NvIop1ixmESYEQNMXYql6NWftBalF8FFkSWM%2BxSQiQE3eZ8lsmc06FNjOwKA4wADiRB%2BYwG%2BVw1BMCMHbOboImAOC8HmILH4Vm5gMTwDLBWBF4xsCglRfWHwJA6VpnMHgGhcLFQIRFK8VwDA4Iw3tgKvwtBEUMBBDi1CAiMat0zGYvMJL5xzwyYwclmJGDsD5ZWTJ9UmwtgAPxyJ-ubPlmQEAiw-LKQVKzhW1MUS7BpCBJpNO5QQXlSiECdJ5EHJAIc%2BnVGtdmE645hmuvdfysu5cFmG0DbDLhBj27IVxWhIsiASjJCfJLDuxKYQWu8dgBFrAC2mqyEa7W-9bnRuhsbB5BTWqOmdDoZIXi3E6BJImaiVq-C4kJFQFKshqAFuEOYUyMjC7SEzoO9gD8R2DrVY2HyNYyR8mBrAMa%2BBcihXILSTIok5QbhIOcWyo8mmUnFViRwagnBkJVUO1gbLsD0w3TRRwA6IAQF3YgAgQSPzMCgHq%2Bg2JtV9sLSOvtlhkz2I1Uo-lKU%2BTEWwPOxg6qvUbmIByNQkE-DqitAAGQMIaa0Op6Qbh4HABgqoMQXvxG5dtdY%2BSgbLXQiDeAzVjyAoTeOQZY0-OFaodQRKFUwho0SOj9ZV18jgAnbGWR2Awj7t7Tlko%2B62x41w-jIiAU5peCJpd9HdWNWJutBusn5Ocf9hBMtdw6F3A6l1CCGLkEacExYqkY4JNGf%2Bd03QMI7MByU3cC2CArY91U7o9TjRNP8TEd3G8BnoUD2Mz6uTfggshZvP5p5sYGEwLgYwxzqznPytcxI4hWL0EmZ8y8HLbA8vYoC1eIhbBmHPzYToArfHIsuZJaVshd5rLQB0DJqrfgatMJYQQNrlysuKlGaUt5Hy91qac114rJLgWlIhYtl8fzhtyZeHN5JC2oVroa4dtpjiOswiK1poTLwNvJPsY4vbMJzvjISFZ9oCiXmlJOeQaZV3qurdu656ZT2OkvYOz95Jf3pmfZs0O5FDL0XLcK8D6L2m3iIrnvStFkPUvY5RXjhrYa%2BWesB6N9H2a7vdrTEo4tHrOmVf26m5%2B4bPX5ONb%2BTVSizV5qteFlbxhuu5oQJahjwgGeb1LczmEPPpfmrF94%2BHSgG2AnjMQfaLZNREgQOYagzwWhlHEFIbg1wONxysPNQ3ZQDE%2BGqBuFTDnWi2-WfbjcfnndG7t9UY3nbhQEHd9zhl6WbZe9d3XUE9u-ddhFEH7LTFGF1cqS78ru1feKJgD4CIGfqAlGz9QjcY2WvGSmzbn332SlHaIttlsNu-nx9m9Di7HTy9u4d88qvLeEgA5dz7mPLY48d-hYT5HNu0-R8zz4AADLn-PERp%2Bz9I2zsnAqjeiqgOKrPw-FTy750rq13v28D9gEPx0a4YDWDqEoNsYBlQIFQIZJY5QZBgGsFyIAA&label=LYewrgzgpg%2BgLuAxgCwogTlKA7GAzEdYIA)

### CODE REVIEW

**Comment by gdevenyi** [comment]
Link not up to date

File: lib/forms/CoBraLab-Mouse-Touchscreen-Form/index.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      variant: 'input',
+      label: 'Chamber serial code'
+    },
+    milkshakeBrand: {
+      kind: 'string',
+      variant: 'radio',
+      label: "Milkshake brand",
+      options: {
+        "Neilson Strawberry Milkshake":"Neilson Strawberry Milkshake",
+        "other": "other"
```
Comment (gdevenyi) [inline lib/forms/CoBraLab-Mouse-Touchscreen-Form/index.ts:107]: capitalization

File: lib/forms/CoBraLab-Mouse-Touchscreen-Form/index.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        }
+        return null
+      }
+    },
+    chamberNumber: {
+      kind: 'number',
+      variant: 'select',
+      label: 'Chamber number',
+      options: {
+        1: "one",
```
Comment (gdevenyi) [inline lib/forms/CoBraLab-Mouse-Touchscreen-Form/index.ts:82]: I don't think we need to mix words and numbers here, its very weird.

Try another variant?

Should this just be a zod-restricted input field instead?

**Review by gdevenyi — CHANGES_REQUESTED** [review]

File: lib/forms/CoBraLab-Mouse-Touchscreen-Form/index.ts (hunk 2 above)
Comment (david-roper) [reply lib/forms/CoBraLab-Mouse-Touchscreen-Form/index.ts:82]: Yeah the select and radio forcing you to have a number label is weird, I will switch it to limited number input instead

**Review by gdevenyi — APPROVED** [review]

**Pull request merged**